pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -310,6 +334,40 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 
+	// Query for StorageClass for AD API if provided
+	shouldRenderADPVC := false
+	if !isManagedCluster && instance.Spec.AnomalyDetection.StorageType == operatorv1.PersistentStorageType {
+		// validate to degrade early if the storage class name is not valid
+		if err = utils.ValidateResourceNameIsQualified(instance.Spec.AnomalyDetection.StorageClassName); err != nil {
+			errMessage := "invalid Anomaly Detection Storage Class name provided"
+			r.SetDegraded(operatorv1.InvalidConfigurationError, errMessage, err, reqLogger)
+			return reconcile.Result{}, err
+		}
+
+		if err = r.client.Get(ctx, client.ObjectKey{Name: instance.Spec.AnomalyDetection.StorageClassName}, &storagev1.StorageClass{}); err != nil {
+			if errors.IsNotFound(err) {
+				errMessage := "failed to get storage class for anomaly detection"
+				r.SetDegraded(operatorv1.ResourceNotFound, errMessage, err, reqLogger)
+				return reconcile.Result{}, err
+			} else {
+				errMessage := "failed to query for Anomaly Detection Storage Class"
+				r.SetDegraded(operatorv1.InternalServerError, errMessage, err, reqLogger)
```

Comment:
I think instead of `InternalServerError` here we should be using `ResourceReadError`